### ISSUE TITLE
IGConv for single GPU? in neosr or basicsr

### ISSUE THREAD

**Issue by seanko29** [dslisleedh/IGConv#1]
I only have a single A100 GPU in our lab.
I want to finetune IGConv on real-world dataset (pretrained mambaIR or HAT). 

1. Is it highly not recommended to use IGConv+ (x4) scale in this case?
2. Will it cause performance degradation? 
3. Do you guys have other suggestions?
I have 2 RTX4090s but they dont have NVLink so I cannot run them in DDP (so it is quite useless I guess).

**Comment by dslisleedh** [comment]
Thank you for your interest, @seanko29 !

For Q1/Q3, I believe you can use a single GPU. We recommend using multiple GPUs to leverage generalized gradients computed from multiple scales. 
For example, GPU0 can compute the x2 loss/gradient while GPU1 computes the x4 loss/gradient, and the master node will average all the gradients. 
Thus, if you are considering only a single scale like x4, a single GPU is sufficient.

For Q2, I'm not sure whether the performance will be better, worse, or remain the same. 
It seems that the performance of IGConv+ at x4 is improved by incorporating x2 or x3, which provides more general information that is lacking in the x4 input. 
So, leveraging a pre-trained model on multiple scales to train x4 with unknown degradations might be helpful; however,  I believe that as training progresses, the useful information learned from x2 and x3 will ultimately be forgotten.

**Comment by seanko29** [comment]
Hi @dslisleedh !! Thank you for the awesome work!

I wanted to finetune the model of HAT model.
The problem is that I think there are issues with the basicsr package (I did not install with pip install. Instead I did python setup.py develop)
I also tried the method in readme file.

In IGConv, it has both basicsr and igconv(which is also a basicsr package format).
I guess anaconda env is crashing as it cannot call some files because of this.

Or do you think there are other issues pertaining to the matter below?


![Image](https://github.com/user-attachments/assets/561104d8-ebc2-4957-91c2-2858e926e795)

This problem also persisted with numpy and scipy (I downgraded these packages).
Are there any other methods to fix this..?

**Comment by dslisleedh** [comment]
It appears that basicsr is being imported from site-packages, which shouldn’t happen if it was installed using `python setup.py develop`. 
Also, the requirements.txt specifies that a downgraded version of numpy (numpy==1.24.1) should be installed.

Please verify that you are activating the correct environment and following the installation instructions in the README.md precisely.
